Obj_To_Stl/src/ObjReader.cpp

```diff
@@ -0,0 +1,93 @@
+#include "../headers/ObjReader.h"
+#include "../headers/Triangulation.h"
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <vector>
+
+Shapes3D::ObjReader::ObjReader() {}
+Shapes3D::ObjReader::~ObjReader() {}
+
+void Shapes3D::ObjReader::read(const std::string &filePath, Triangulation &triangulationObj)
+{
+    std::ifstream file(filePath);
+
+    if (!file.is_open())
+    {
+        std::cout << "Error while opening .obj file." << std::endl;
+        return;
+    }
+
+    std::string line;
+    std::vector<Point3D> points;
+    std::vector<Triangle> triangles;
+    std::vector<Point3D> normals;
+
+    // reads obj file line by line
+    while (getline(file, line))
+    {
+        std::istringstream vertexStream(line);
+        std::string keyword;
+        vertexStream >> keyword;
+
+        if (keyword == "v")
+        {
+            double x, y, z;
+            vertexStream >> x >> y >> z;
+            Point3D point(x, y, z);
+            triangulationObj.uniquePoints().push_back(point);
+        }
+        else if (keyword == "vn")
+        {
+            double x, y, z;
+            vertexStream >> x >> y >> z;
+            Point3D normal(x, y, z);

```

Comment:
same as point

Reply:
As we require Point3D (points) and int (indices) for mapping, these both are included in Triangulation class. The same goes with Point3D (normal) and int (indices).